### PR TITLE
planner: add AGG_TO_COP hint

### PR DESCRIPTION
**Pull request by foreyes**
### What problem does this PR solve? 

Add sql hint for enforcing push aggregation to coprocessor.

### What is changed and how it works?

Check if the logical aggregation can be push down first, and throw a warning if not.

Then only cop task will be considered when enumerating task types with AGG_TO_COP hint.

### Check List 

Tests 

 - Unit test

Code changes

 - None

Side effects

 - None

Related changes

 - None

### CODE REVIEW

File: planner/core/exhaust_physical_plans.go [hunk 1]
```diff
@@ -1266,6 +1266,12 @@ func (p *baseLogicalPlan) exhaustPhysicalPlans(_ *property.PhysicalProperty) []P
 	panic("baseLogicalPlan.exhaustPhysicalPlans() should never be called.")
 }
 
+func (la *LogicalAggregation) canPushDown() bool {
```
Comment (lzmhhh123) [inline planner/core/exhaust_physical_plans.go]:
```suggestion
func (la *LogicalAggregation) canPushToCop() bool {
```

File: planner/core/exhaust_physical_plans.go [hunk 2]
```diff
@@ -1266,6 +1266,12 @@ func (p *baseLogicalPlan) exhaustPhysicalPlans(_ *property.PhysicalProperty) []P
...
+	// Child may be of other types in the future, such as limit, topn and projection.
```
Comment (eurekaka) [inline planner/core/exhaust_physical_plans.go]: Please add more comments to explain why we cannot push aggregation down to coprocessor if the child is not DataSource. For example, the check must be called after logical optimization, and current pushed down `Limit` / `TopN` would convert task to rootTask, etc. Also, add a TODO to remind that we should check `Projection` as well if we support `Projection` pushdown later.

Comment (foreyes) [reply planner/core/exhaust_physical_plans.go]: Okay.

**Comment by foreyes** [comment]
Accept suggestion and change some test, PTAL @lzmhhh123 
Address comment, PTAL @eurekaka

File: planner/core/exhaust_physical_plans.go [hunk 3]
```diff
@@ -1271,6 +1271,16 @@ func (p *baseLogicalPlan) exhaustPhysicalPlans(_ *property.PhysicalProperty) []P
 	panic("baseLogicalPlan.exhaustPhysicalPlans() should never be called.")
 }
 
+func (la *LogicalAggregation) canPushToCop() bool {
+	// At present, only Aggregation, Limit, TopN can be push to cop task, and Projection will be supported in the future.
```
Comment (eurekaka) [inline planner/core/exhaust_physical_plans.go]:
```suggestion
	// At present, only Aggregation, Limit, TopN can be pushed to cop task, and Projection will be supported in the future.
```

**Pull request merged**